feat(minifier): support subtraction assignment.

7086cmd opened the pull request:
Due to the potential for string concatenation when using the `+=` operator, we should only handle the scenario when using the `-=` operator.

Boshen commented on crates/oxc_minifier/src/ast_passes/peephole_substitute_alternate_syntax.rs:
```diff
@@ -260,6 +267,37 @@ impl<'a> PeepholeSubstituteAlternateSyntax {
+    ) -> Option<Expression<'a>> {
+        let Some(target) = expr.left.as_simple_assignment_target() else { return None };
+        // TODO avoid using `clone_in`, as it is performance-intensive
+        let target = target.clone_in(ctx.ast.allocator);
```
I think you should only clone when necessary, this is currently clone on every assignment expression?

7086cmd replied:
The clone only appears in the following situations:
- Is the target situation (from `-= 1` or `-= -1`)
- Is a simple assignment target
I am also investigating whether we can use `mem::replace`.

7086cmd replied:
I used the transform, and it is solved.

7086cmd commented on crates/oxc_minifier/src/ast_passes/peephole_fold_constants.rs:
```diff
@@ -1,3 +1,14 @@
+use std::cmp::Ordering;
+
+use num_bigint::BigInt;
+use oxc_ast::ast::*;
+use oxc_span::{GetSpan, Span, SPAN};
+use oxc_syntax::{
+    number::{NumberBase, ToJsInt32},
+    operator::{BinaryOperator, LogicalOperator, UnaryOperator},
+};
+use oxc_traverse::{Ancestor, Traverse, TraverseCtx};
+
```
This resolves a prev problem in https://github.com/oxc-project/oxc/pull/6205#discussion_r1782490192.

7086cmd committed:
feat(minifier): support subtraction assignment. (#6214)

Due to the potential for string concatenation when using the `+=` operator, we should only handle the scenario when using the `-=` operator.
